packages/state/source/wallets/wallet-repository-clone.ts

```diff
@@ -45,7 +45,7 @@ export class WalletRepositoryClone extends WalletRepository implements Contracts
 		if (!super.hasByIndex(Contracts.State.WalletIndexes.PublicKeys, publicKey)) {
 			const wallet = this.findByAddress(await this.addressFactory.fromPublicKey(publicKey));
 			wallet.setPublicKey(publicKey);
-			super.index(wallet);
+			this.getIndex(Contracts.State.WalletIndexes.PublicKeys).set(publicKey, this.findHolder(wallet));
```

Comment:
I think we can use `setOnIndex` here.

Reply:
Can be, but it is a subject of change in the upcoming PRs.